firmware/examples/smoltcp-example/src/time.rs

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: 2022 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use smoltcp::time::{Duration, Instant};
+
+#[derive(Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct Clock{
+    global_time: u64,
+    timer_last: u32,
+    ticks_per_micro: u64,
+    timer_now: *const u32
+}
+
+impl Clock {
+    pub fn new(addr: usize, frequency: u64) -> Clock {
```

Comment:
Since `addr` is meant to be the value of a pointer, this should be `addr: *mut u32` so it's clear to the caller.

Also because this function blindly dereferences that point without any security checks this whole function should be `unsafe`. Since this constructs a hardware abstraction it should be unsafe, but other methods on it don't need to be unsafe :)